datastore/mock_store.go

```diff
@@ -0,0 +1,133 @@
+package datastore
```

Comment:
This mock_store is purely there for testing. Isn't it? If so can we rename this file to mock_store_test.go so that it doesn't get built with production package?